packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -30,42 +30,44 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
   const [slices, setSlices] = useState<Slice[]>([]);
   const [isCreatingSlices, setIsCreatingSlices] = useState(false);
 
+  const setSlice = (index: number, sliceUpdate: Slice) => {
+    setSlices((slices) =>
+      slices.map((s, i) => (i === index ? { ...s, ...sliceUpdate } : s)),
+    );
+  };
+
+  const uploadImage = (index: number, image: File) => {
```

Comment:
here too we should avoid positional arguments

Reply:
Done ☑️